### PR TITLE
Avoid redundant Map lookups

### PR DESCRIPTION
**Pull request by XN137**
while reading through the code i noticed a few redundant map lookups

i've only adjusted the places where it was clear the map never stores null values and where `get` immediately followed a `containsKey`

according to running the JMH benchmarks locally this safes a few cycles and often makes the code clearer imo

### CODE REVIEW

File: nullaway/src/main/java/com/uber/nullaway/NullAway.java [hunk 1]
```diff
@@ -2412,11 +2411,8 @@ private static ExpressionTree stripParensAndCasts(ExpressionTree expr) {
    * @return computed nullness for e, if any, else Nullable
    */
   public Nullness getComputedNullness(ExpressionTree e) {
-    if (computedNullnessMap.containsKey(e)) {
-      return computedNullnessMap.get(e);
-    } else {
-      return Nullness.NULLABLE;
-    }
+    Nullness nullness = computedNullnessMap.get(e);
```
Comment (XN137) [inline nullaway/src/main/java/com/uber/nullaway/NullAway.java:2414]: wanted to use `getOrDefault` here, but ran into https://github.com/uber/NullAway/issues/723

**Comment by msridhar** [comment]
> according to running the JMH benchmarks locally this safes a few cycles and often makes the code clearer imo

Does this really show a JMH benchmark improvement locally that is outside the margin of error?

File: nullaway/src/main/java/com/uber/nullaway/NullAway.java (hunk 1 above)
Comment (msridhar) [reply nullaway/src/main/java/com/uber/nullaway/NullAway.java:2414]: Can we add a comment here to use `getOrDefault` once it is supported properly?

**Review by msridhar — COMMENTED** [review]
These changes overall LGTM!  Thanks for this!  Just one minor comment

File: nullaway/src/main/java/com/uber/nullaway/NullAway.java [hunk 2]
```diff
@@ -1208,12 +1209,10 @@ private boolean fieldInitializedByPreviousInitializer(
...
     ClassTree enclosingClass = (ClassTree) enclosingClassPath.getLeaf();
+    ClassSymbol classSymbol = ASTHelpers.getSymbol(enclosingClass);
     Multimap<Tree, Element> tree2Init =
-        initTree2PrevFieldInit.get(ASTHelpers.getSymbol(enclosingClass));
-    if (tree2Init == null) {
-      tree2Init = computeTree2Init(enclosingClassPath, state);
-      initTree2PrevFieldInit.put(ASTHelpers.getSymbol(enclosingClass), tree2Init);
-    }
+        initTree2PrevFieldInit.computeIfAbsent(
+            classSymbol, sym -> computeTree2Init(enclosingClassPath, state));
```
Comment (lazaroclapp) [inline nullaway/src/main/java/com/uber/nullaway/NullAway.java:1215]: Surprised that `computeIfAbsent` is measurably faster than `get` -> null check -> `put`, _but_ it's more readable, so 🙂

**Review by lazaroclapp — COMMENTED** [review]
Other than the need to rebase and Manu's comment, this generally LGTM. If you can share the local JMH numbers, that might be useful. I am surprised this makes a measurable performance difference, but OTOH, I agree that it's overall cleaner.

File: nullaway/src/main/java/com/uber/nullaway/NullAway.java (hunk 1 above)
Comment (XN137) [reply nullaway/src/main/java/com/uber/nullaway/NullAway.java:2415]: added but this pattern exists in a few other places as well, i had to undo my changes to them (see first force-push on this PR)

**Comment by XN137** [comment]
> Other than the need to rebase and Manu's comment, this generally LGTM. 

it was based on the latest master commit at the point of creating the PR, but ok rebased again and added the comment.

> I am surprised this makes a measurable performance difference

i can try to post local JMH numbers (but was hoping you can put a label on the PR and then compare to last master run).
those are micro benchmarks after all, so things like not having to hash the key twice should be "noticable" (i only said it should "safe a few cycles" :sweat_smile:)

**Comment by msridhar** [comment]
> i can try to post local JMH numbers (but was hoping you can put a label on the PR and then compare to last master run). those are micro benchmarks after all, so things like not having to hash the key twice should be "noticable" (i only said it should "safe a few cycles" 😅)

I totally believe this could save a few cycles.  I just never had any luck showing a measurable benefit from a small improvement like that using our JMH setup.  Due to noise from the JVM warmup, a not fully idle machine, etc., usually the error bars are large enough that such a small improvement would be in the noise.  If you actually saw a perf improvement that was outside the noise that would be cool!  Of course I can try to measure later, doesn't need to hold up this PR.

**Comment by XN137** [comment]
>  If you actually saw a perf improvement that was outside the noise that would be cool!

numbers were slightly better for a bunch of local runs but i agree there was a lot of noise as well...
feel free to measure this yourself but imo it should be somewhat clear why this saves a few cycles or at least cant make things worse. with my JMH remark i was not claiming this fixes a huge performance bottleneck or the like, the change is mainly about more concise code imo.

to me this combination usually suggests that `null` is a valid map value:
```
if (map.containsKey(k)) {
  V v = map.get(k);
}
```
because why not get the value immediately otherwise.

**Review by msridhar — APPROVED** [review]

**Comment by msridhar** [comment]
Thanks for the contribution again!

**Comment by XN137** [comment]
CI on master seems to be failing, feel free to revert if it caused by my changes somehow (i cant see the action logs to investigate)

**Comment by msridhar** [comment]
Not your change; there is some other system issue.  Thanks for checking!

**Pull request merged**